data_structures/src/data_request.rs

```diff
@@ -467,16 +506,34 @@ where
 
         // Collateral division rest goes for the miner
         let non_reveals_count = commits_count - reveals_count;
-        let (reward, _rest) = calculate_witness_reward(
-            commits_count,
-            reveals_count,
-            liars_count + non_reveals_count,
-            errors_count,
-            dr_output.witness_reward,
-            collateral,
-        );
+        let is_after_second_hard_fork = after_second_hard_fork(block_epoch, get_environment());
+        let (reward, _rest) = if is_after_second_hard_fork {
+            calculate_witness_reward(
+                commits_count,
+                liars_count + non_reveals_count,
+                errors_count,
+                dr_output.witness_reward,
+                collateral,
+            )
+        } else {
+            calculate_witness_reward_before_second_hard_fork(
+                commits_count,
+                reveals_count,
+                liars_count + non_reveals_count,
+                errors_count,
+                dr_output.witness_reward,
+                collateral,
+            )
+        };
```

Comment:
At some point in the future, this code can be refactored by calling a `calculate_witness_reward` function with the block_epoch as argument.